intera_examples/scripts/constrained_zeroG.py

```diff
@@ -24,16 +24,168 @@
+            unconstrained_axes = free_axis_list
+
+        # set the stiffness to zero by default
+        interaction_options.set_K_impedance([0, 0, 0, 0, 0, 0])
+
+        # set the axes with maximum stiffness
+        interaction_options.set_max_impedance(boolToggle(int2bool(unconstrained_axes)))
+
+        interaction_options.set_in_endpoint_frame(args.in_endpoint_frame)
+
+        # set nullspace stiffness to zero if nullspace_only option is provided
+        if args.nullspace_only:
+            K_nullspace = [0, 0, 0, 0, 0, 0, 0]
+        else:
+            K_nullspace = args.K_nullspace
+
+        interaction_options.set_K_nullspace(K_nullspace)
+
+        if len(args.interaction_frame) < 7:
+            rospy.logerr('The number of elements must be 7!')
+        elif len(args.interaction_frame) == 7:
+            quat_sum_square = (args.interaction_frame[3]*args.interaction_frame[3]
+                              + args.interaction_frame[4]*args.interaction_frame[4]
+                              + args.interaction_frame[5]*args.interaction_frame[5]
+                              + args.interaction_frame[6]*args.interaction_frame[6])
+            if quat_sum_square  < 1.0 + 1e-7 and quat_sum_square > 1.0 - 1e-7:
+                interaction_frame = Pose()
+                interaction_frame.position.x = args.interaction_frame[0]
+                interaction_frame.position.y = args.interaction_frame[1]
+                interaction_frame.position.z = args.interaction_frame[2]
+                interaction_frame.orientation.w = args.interaction_frame[3]
+                interaction_frame.orientation.x = args.interaction_frame[4]
+                interaction_frame.orientation.y = args.interaction_frame[5]
+                interaction_frame.orientation.z = args.interaction_frame[6]
+                interaction_options.set_interaction_frame(interaction_frame)
+            else:
+                rospy.logerr('Invalid input to quaternion! The quaternion must be a unit quaternion!')
+        else:
+            rospy.logerr('Invalid input to interaction_frame!')
+
+        # always enable the rotations for constrained zero-G
+        interaction_options.set_rotations_for_constrained_zeroG(True)
+
+        self.msg = interaction_options.to_msg()
+
+    def send_msg(self):
+        try:
+            # print the resultant interaction options once
+            rospy.loginfo(self.msg)
+            self.pub.publish(self.msg)
+            rs = intera_interface.RobotEnable(CHECK_VERSION)
+            if self.repeat:
+                while not rospy.is_shutdown() and rs.state().enabled:
+                    self.rate.sleep()
+                    self.pub.publish(self.msg)
+        except rospy.ROSInterruptException:
+            rospy.logerr('Keyboard interrupt detected from the user. %s',
+                         'Exiting the node...')
+        if not rs.state().enabled:
+            self.send_position_cmd()
+
+    def send_position_cmd(self):
```

Comment:
Perhaps change the function name to: `send_position_mode_cmd`. This would make it more clear that we are switching the interaction mode to position, rather than just updating the position of the robot.

Reply:
Good call